### PR TITLE
Json test

### PR DESCRIPTION
**Pull request by gearama**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+          }
+          case Action::Deserialize: {
+            m_testObject.Deserialize(m_jsonBody);
+            break;
+          }
+        }
+      }
+      catch (std::exception const&)
+      {
+        // don't print exceptions, they are happening at each request, this is the point of the test
```
Comment (LarryOsterman) [inline sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp:296]: I'm not sure this is valid for this test. Are you really ok with the test swallowing exceptions? And if they are thrown, what does throwing the exception do to the overall measurement? Doesn't throwing an exception introduce a large amount of noise into the results depending on whether it is thrown or not?

File: sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+      NullableInt64 = static_cast<int64_t>(4);
+      NullableUint8 = static_cast<uint8_t>(5);
+      NullableUint16 = static_cast<uint16_t>(6);
+      NullableUint32 = static_cast<uint32_t>(7);
+      NullableUint64 = static_cast<uint64_t>(8);
+      NullableFloat = static_cast<float>(9.0);
+      NullableDouble = static_cast<double>(10.0);
+      NullableString = "string";
+
+      Booleans = std::vector<bool>(vectorSize, true);
```
Comment (LarryOsterman) [inline sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp:215]: Does it make sense to break this down to a finer granularity?

File: sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp (hunk 1 above)
Comment (gearama) [reply sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp:296]: sure, makes sense

File: sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp (hunk 2 above)
Comment (gearama) [reply sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp:215]: i'm not sure there is a lot of benefit breaking it down further, once i start profiling, if it makes sense to i will break it down, at the end of the day this code is not set in stone .

File: sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp (hunk 1 above)
Comment (LarryOsterman) [reply sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp:296]: Btw, in other cases, I'm a fan of catching exceptions and doing something, but for a perf test, we need to make sure that exceptions aren't thrown (unless we're testing the performance overhead of exceptions ::).

**Review by LarryOsterman — APPROVED** [review]

**Review by RickWinter — APPROVED** [review]

**Pull request merged**